language/testing-infra/functional-tests/src/evaluator.rs

```diff
@@ -574,7 +575,20 @@ fn eval_transaction<TComp: Compiler>(
                     log.append(EvaluationOutput::Output(OutputType::CompilerLog(s)));
                 }
             };
-            unwrap_or_abort!(compiler.compile(compiler_log, sender_addr, &transaction.input))
+            static FILENAME_PAT: Lazy<Regex> =
+                Lazy::new(|| Regex::new(r"[A-Za-z0-9_/.]+:([0-9]+):([0-9]+)").unwrap());
+
+            let compiler_res = compiler.compile(compiler_log, sender_addr, &transaction.input);
+            let filtered_compiler_res = compiler_res.map_err(|err| {
+                let msg = err.root_cause().to_string();
```

Comment:
Does root_cause just grab the top level message? By that I mean does it discard the secondary labels?

Reply:
It's supposed to grab the deepest (aka "root") error in the chain. However looks like it's not needed here. Will get it removed.